azure-mgmt-batch/src/main/java/com/microsoft/azure/management/batch/implementation/ApplicationPackagesInner.java

```diff
@@ -276,7 +276,7 @@ public ApplicationPackageInner call(ServiceResponse<ApplicationPackageInner> res
         if (this.client.apiVersion() == null) {
             throw new IllegalArgumentException("Parameter this.client.apiVersion() is required and cannot be null.");
         }
-        return service.create(resourceGroupName, accountName, applicationName, versionName, this.client.subscriptionId(), this.client.apiVersion(), this.client.acceptLanguage(), parameters, this.client.userAgent())
+        return service.create(resourceGroupName, accountName, applicationName, versionName, this.client.subscriptionId(), this.client.apiVersion(), this.client.acceptLanguage(), null, this.client.userAgent())
```

Comment:
Why it is null?

Reply:
Fixed in later commit by using default constructor.

Reply:
But this should be the parameter that need to be passed in, isn't it?

Reply:
No, according to above links, only name is required rather than payload. And double checked how we define a new package [here](https://github.com/Azure/azure-libraries-for-java/blob/master/azure-mgmt-batch/src/main/java/com/microsoft/azure/management/batch/Application.java#L81)

Reply:
OK, I see. All its properties is readonly. Maybe this is the reason this object is not in parameter.